spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubOperations.java

```diff
@@ -42,50 +40,19 @@
 	/**
 	 * Send a message to Pub/Sub.
 	 * @param topic the name of an existing topic
-	 * @param payload the message String payload
-	 * @param headers map of String to String headers
-	 * @return the listenable future of the call
-	 */
-	ListenableFuture<String> publish(String topic, String payload, Map<String, String> headers);
-
-	/**
-	 * Send a message to Pub/Sub.
-	 * @param topic the name of an existing topic
-	 * @param payload the message String payload
-	 * @param headers map of String to String headers
-	 * @param charset charset to decode the payload
-	 * @return the listenable future of the call
-	 */
-	ListenableFuture<String> publish(String topic, String payload, Map<String, String> headers,
-			Charset charset);
-
-	/**
-	 * Send a message to Pub/Sub.
-	 * @param topic the name of an existing topic
-	 * @param payload the message payload in bytes
-	 * @param headers map of String to String headers
-	 * @return the listenable future of the call
-	 */
-	ListenableFuture<String> publish(String topic, byte[] payload, Map<String, String> headers);
-
-	/**
-	 * Send a message to Pub/Sub.
-	 * @param topic the name of an existing topic
-	 * @param payload the message payload on the {@link PubsubMessage} payload format
-	 * @param headers map of String to String headers
+	 * @param payload an object that will be serialized and sent
 	 * @return the listenable future of the call
 	 */
-	ListenableFuture<String> publish(String topic, ByteString payload, Map<String, String> headers);
+	<T> ListenableFuture<String> publish(String topic, T payload,
+			Map<String, String> headers) throws IOException;
```

Comment:
Can't we don't expose this exception as a public contract in this method?
May we wrap into the `UncheckedIOException` whenever an `IOException` is thrown in the target implementation?

We can mention `UncheckedIOException` in the JavaDocs though for better behavior understanding.

What worries me here that we not always get an `IOException` in the impl, therefore no reason to enforce end-user to always handle it.

Does it make sense to you?
Thanks